cpp/daal/src/data_management/finiteness_checker_cpu.cpp

```diff
@@ -0,0 +1,320 @@
+
+    daal::TlsMem<bool, cpu, services::internal::ScalableCalloc<bool, cpu> > tlsNotFinite(1);
+
+    daal::threader_for_break(nCols, nCols, [&](size_t i, bool & needBreak) {
+        bool * localNotFinite = tlsNotFinite.local();
+        DAAL_CHECK_MALLOC_THR(localNotFinite);
+
+        switch ((*tableFeaturesDict)[i].getIndexType())
+        {
+        case daal::data_management::features::IndexNumType::DAAL_FLOAT32:
+        {
+            ReadColumns<float, cpu> colBlock(table, i, 0, nRows);
+            DAAL_CHECK_BLOCK_STATUS_THR(colBlock);
+            const float * colPtr = colBlock.get();
+            // defined in finiteness_checker_avx2_impl.i and finiteness_checker_avx512_impl.i for avx2 and avx512 respectively
+            *localNotFinite |= !checkFiniteness<float, cpu>(nRows, 1, nRows, &colPtr, allowNaN);
+            break;
+        }
+        case daal::data_management::features::IndexNumType::DAAL_FLOAT64:
+        {
+            ReadColumns<double, cpu> colBlock(table, i, 0, nRows);
+            DAAL_CHECK_BLOCK_STATUS_THR(colBlock);
+            const double * colPtr = colBlock.get();
+            *localNotFinite |= !checkFiniteness<double, cpu>(nRows, 1, nRows, &colPtr, allowNaN);
+            break;
+        }
+        default: break;
+        }
+
+        if (*localNotFinite)
+        {
+            needBreak = true;
+            breakFlag = true;
+        }
+    });
+
+    st |= safeStat.detach();
+    if (!st)
+    {
+        return false;
+    }
+
+    if (breakFlag)
+    {
+        valuesAreFinite = false;
+    }
+    else
+    {
+        tlsNotFinite.reduce([&](bool * localNotFinite) { valuesAreFinite &= !*localNotFinite; });
+    }
+
+    return valuesAreFinite;
+}
+
+    #if (__CPUID__(DAAL_CPU) == __avx512__)
+
+        #include "finiteness_checker_avx512_impl.i"
+
+    #endif // __CPUID__(DAAL_CPU) == __avx512__
+    #if (__CPUID__(DAAL_CPU) == __avx2__)
+
+        #include "finiteness_checker_avx2_impl.i"
+
+    #endif // __CPUID__(DAAL_CPU) == __avx2__
+
+#endif
+
+template <typename DataType, daal::CpuType cpu>
+services::Status allValuesAreFiniteImpl(NumericTable & table, bool allowNaN, bool * finiteness)
+{
+    services::Status s;
+    const size_t nRows    = table.getNumberOfRows();
+    const size_t nColumns = table.getNumberOfColumns();
+    DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(size_t, nRows, nColumns);
+    const size_t nElements = nRows * nColumns;
+    const NTLayout layout  = table.getDataLayout();
+
+    // first stage: compute sum of all values and check its finiteness
+    double sum       = 0;
+    bool sumIsFinite = true;
+
+    if (layout == NTLayout::soa)
+    {
+        sum = computeSumSOA<cpu>(table, sumIsFinite, s);
+    }
+    else
+    {
+        ReadRows<DataType, cpu> dataBlock(table, 0, nRows);
+        DAAL_CHECK_BLOCK_STATUS(dataBlock);
+        const DataType * dataPtr = dataBlock.get();
+
+        sum = computeSum<DataType, cpu>(1, nElements, &dataPtr);
```

Comment:
Another FYI. No action is required.

It is better in terms of both performance and memory consumption to move ReadRows inside the parallel for that happens in computeSum.
Because ReadRows in some cases might copy the whole table and convert it -> +1 pass through the memory.
When ReadRows is made by blocks and is combined with the computational part it is usually more efficient.